src/bci_build/package/kiosk.py

```diff
@@ -0,0 +1,165 @@
+                xorg_server_re,
+                package_name="xorg-x11-server",
+                parse_version=ParseVersion.MINOR,
+            )
+        ],
+        extra_files=_X11_FILES,
+        entrypoint=["/usr/local/bin/entrypoint.sh"],
+        custom_end=generate_package_version_check(
+            "xorg-x11-server", tag_ver, ParseVersion.MAJOR
+        )
+        + """
+RUN useradd -m user -u 1000
+COPY preferences /etc/icewm/preferences
+COPY xinitrc /etc/X11/xinit/xinitrc
+COPY xorg.conf /etc/X11/xorg.conf.d/xorg.conf
+
+ENV XDG_SESSION_TYPE=x11
+
+COPY entrypoint.sh /usr/local/bin/entrypoint.sh
+RUN chmod +x /usr/local/bin/entrypoint.sh
+""",
+    )
+    for os_version in ALL_NONBASE_OS_VERSIONS
+]
+
+
+_PULSE_FILES = {
+    "client.conf": ((_pa_dir := _pkg_dir / "pulseaudio") / "client.conf").read_text(),
+    "daemon.conf": (_pa_dir / "daemon.conf").read_text(),
+    "system.pa": (_pa_dir / "system.pa").read_text(),
+}
+
+PULSEAUDIO_CONTAINERS = [
+    ApplicationStackContainer(
+        name="pulseaudio",
+        os_version=os_version,
+        tag_version=(tag_ver := "17"),
+        version_in_uid=False,
+        is_latest=os_version in CAN_BE_LATEST_OS_VERSION,
+        version=(pulseaudio_ver_re := "%%pulseaudio_ver%%"),
+        pretty_name="Pulseaudio",
+        package_list=["pulseaudio", "pulseaudio-utils"],
+        replacements_via_service=[
+            Replacement(
+                pulseaudio_ver_re,
+                package_name="pulseaudio",
+                parse_version=ParseVersion.MINOR,
+            )
+        ],
+        extra_files=_PULSE_FILES,
+        custom_end=generate_package_version_check(
+            "pulseaudio", tag_ver, ParseVersion.MAJOR
+        )
+        + """
+COPY daemon.conf /etc/pulse/
+COPY client.conf /etc/pulse/
+COPY system.pa /etc/pulse/
+""",
+    )
+    for os_version in ALL_NONBASE_OS_VERSIONS
+]
+
+
+FIREFOX_CONTAINERS = [
+    DevelopmentContainer(
+        name="firefox",
+        os_version=os_version,
+        tag_version=(ff_ver := get_pkg_version("MozillaFirefox", os_version)),
+        is_latest=os_version in CAN_BE_LATEST_OS_VERSION,
+        version=(ff_ver_re := "%%ff_ver%%"),
+        version_in_uid=False,
+        pretty_name="Mozilla Firefox",
+        package_list=(
+            ["MozillaFirefox"]
+            + (
+                ["MozillaFirefox-branding-openSUSE"]
+                if os_version.is_tumbleweed
+                else [
+                    "MozillaFirefox-branding-SLE",
+                    # required by Firefox via /usr/bin/gconftool-2
+                    # to be fixed via FileProvides
+                    "gconf2",
+                ]
+            )
+        ),
+        replacements_via_service=[
+            Replacement(
+                ff_ver_re,
+                package_name="MozillaFirefox",
+                parse_version=ParseVersion.MINOR,
+            )
+        ],
+        cmd=["/bin/bash", "-c", "firefox --kiosk $URL"],
```

Comment:
does this work without `$URL` being set? why do we need to pipe this through `bash -c` ?

Reply:
It might be worth including an entrypoint.sh that defaults $URL to something sane. I had played with this a couple weeks ago when I was wondering about attaching devtools remotely: https://build.opensuse.org/projects/home:atgracey:wallboardos/packages/firefox-dev-entrypoint/files/entrypoint.sh?expand=1